index.js

```diff
@@ -55,6 +62,11 @@
   }
   extend(Nothing, Maybe);
```

Comment:
Don't know if this helps, but we solved some of those issues above for fantasy-land implementions (option, either and etc) with this... https://github.com/puffnfresh/daggy/blob/master/src/daggy.js#L103